src/firmware/app/primitives/move_primitive.c

```diff
@@ -1,257 +1,31 @@
-        float abs_final_rotation = fabsf(min_angle_delta(initial_rotation, final_angle));
-        // if we have found a smaller angle, then update the minimum rotation
-        // and chosen index
-        if (abs_final_rotation < minimum_rotation)
-        {
-            minimum_rotation = abs_final_rotation;
-            min_index        = i;
-        }
-    }
-    return min_index;
-}
-
-/**
- * If we are far enough away from our destination, then we should try
- * rotating onto a wheel axis so that we can move faster. We should
- * pick the wheel axis that minimizes the distance the bot will have
- * to rotate to get to its destination angle after rotating onto an
- * axis.
- *
- * @param optimal_wheel_axes_index The index of the wheel axes to rotate onto
- * @param pb The data container that contains information about
- * the direction the robot will move along.
- * @param angle The angle that the robot is currently facing
- */
-void choose_rotation_destination(unsigned optimal_wheel_axes_index, PhysBot* pb,
-                                 float angle)
-{
-    // if we are close enough then we should just allow the bot to rotate
-    // onto its destination angle, so skip this if block
-    if (fabsf(pb->maj.disp) > APPROACH_LIMIT)
-    {
-        float wheel_axes[8];
-        build_wheel_axes(&wheel_axes, angle);
-        float theta_norm = atan2f(pb->pos[1], pb->pos[0]);
-        // use the pre-determined wheel axis
-        pb->rot.disp = min_angle_delta(wheel_axes[optimal_wheel_axes_index], theta_norm);
-    }
-}
-
-
-void plan_move_rotation(PhysBot* pb, float avel)
-{
-    float time_target = (pb->maj.time > TIME_HORIZON) ? pb->maj.time : TIME_HORIZON;
-    if (time_target > 0.5f)
-    {
-        time_target = 0.5f;
-    }
-    pb->rot.time  = time_target;
-    pb->rot.vel   = pb->rot.disp / pb->rot.time;
-    pb->rot.accel = (pb->rot.vel - avel) / TIME_HORIZON;
-    limit(&pb->rot.accel, MAX_T_A);
-}
-
-void app_move_primitive_start(TbotsProto_PrimitiveParams params, void* void_state_ptr,
-                              FirmwareWorld_t* world)
+void app_move_primitive_start(TbotsProto_MovePrimitive prim_msg, void *void_state_ptr,
+                              FirmwareWorld_t *world)
 {
-    MovePrimitiveState_t* state = (MovePrimitiveState_t*)void_state_ptr;
-    // Parameters:     destination_x [mm]
-    //                destination_y [mm]
-    //                destination_ang [centi-rad]
-    //                end_speed [millimeter/s]
-
-    // Convert into m/s and rad/s because physics is in m and s
-    state->destination[0] = params.parameter1 / 1000.0f;
-    state->destination[1] = params.parameter2 / 1000.0f;
-    state->destination[2] = params.parameter3 / 100.0f;
-    state->end_speed      = params.parameter4 / 1000.0f;
-    state->slow           = params.slow;
+    app_move_helper_start(void_state_ptr, world, prim_msg.position_params,
```

Comment:
why can't `move_helper` be a `move_motion`, its very similar to how we handle `chick_motion`?
I think there is a slight inconsistency between `move_helper` and `chick_motion`